src/backend/app/Listeners/PaymentListener.php

```diff
@@ -86,11 +86,11 @@ public function onPaymentSuccess(
                 $this->applianceRatePaymentHistoryService->assign();
                 break;
             case $paidFor instanceof Asset:
-                $paymentHistory->paid_for_type = Asset::class;
+                $paymentHistory->paid_for_type = 'appliance';
```

Comment:
This looks like a massive change. Probably it's correct, but I'm curious: How did you test this?

Reply:
I checked on the frontend, the Meter Transanction list has a condition to display approriate message based on the paid for type. 


```js
     <md-table-row v-for="token in transactions.tokens" :key="token.id">
            <md-table-cell
              v-text="token.transaction.original_transaction_type"
            ></md-table-cell>
            <md-table-cell
              v-text="moneyFormat(token.transaction.amount)"
            ></md-table-cell>
            <md-table-cell v-if="token.paid_for_type === 'token'">
              Token {{ token.paid_for.token }}
            </md-table-cell>
            <md-table-cell v-else>
              {{ token.paid_for_type }}
            </md-table-cell>
            <md-table-cell
              v-if="token.paid_for_type === 'token'"
              v-text="readable(token.paid_for.energy) + 'kWh'"
            ></md-table-cell>
            <md-table-cell v-else>-</md-table-cell>
            <md-table-cell
              v-text="timeForTimeZone(token.created_at)"
            ></md-table-cell>
          </md-table-row>
  ```
  
  However, on the backend, the string is updated to the internal class name (i.e `App\Model\Token`), which did not match with what the frontend expected; I found that weird. 
  
  Also, while searching for the usecases for `paid_for_type` field I found this factory. The values seem to tally with the frontend expectation
  
  ```php
  class PaymentHistoryFactory extends Factory {
    protected $model = PaymentHistory::class;

    /**
     * Define the model's default state.
     *
     * @return array
     */
    public function definition() {
        return [
            'id' => 1,
            'transaction_id' => 1,
            'amount' => $this->faker->randomFloat(2, 0, 100),
            'payment_service' => $this->faker->randomElement(['vodacom_transaction', 'airtel_transaction', 'agent_transaction']),
            'sender' => $this->faker->phoneNumber,
            'payment_type' => $this->faker->randomElement(['appliance', 'energy', 'installment', 'access rate']),
            'paid_for_type' => $this->faker->randomElement(['appliance', 'token', 'loan_rate', 'access_rate']),
            'paid_for_id' => 1,
            'payer_type' => 'person',
            'payer_id' => 1,
        ];
    }
}
```

Reply:
What are your thoughts? Do you think we should have a different display for tokens? Right now the frontend also gets the token data we could potentially use for that (from the Meter Token table)

Reply:
There are two questions here. Let me address them individually:

**What should be the value for `paid_for`?** `App\Model\Token` vs `token`

If I understand correctly `paid_for` defines a [Polymorphic relation ship](https://laravel.com/docs/11.x/eloquent-relationships#one-of-many-polymorphic-relations). I would like to follow the practise that is standard in Laravel. On Database level it I think it should be the class type `App\Model\Token` as per Laravel convention of defining polymorphic relations (but I'm also not a Laravel expert, maybe I got this wrong), rather than the display name `token`.

So, I'd suggest we adapt the frontend filter to check for `App\Model\Token`.

**How should token be displayed?**

My preferred solution would be to adapt the Frontend to display Token objects like this:

`Token (1234-1234-1234)` or even `Token (123 412 341 234)` (this would also resolve https://github.com/EnAccess/micropowermanager/issues/83)

But this should be a pure frontend change. I.e. it should change the tokens in the backend, but just the way they are presented to the user

WDYT?

Reply:
After reading up on how Laravel handles Polymorphic relationship, I now understand your explanations.  

I will revert this and make the frontend change as suggested. Thank you @dmohns